### PR TITLE
RGB images support

### PR DESCRIPTION
**Pull request by rahulbshrestha**
This PR fixes the issue mentioned in https://github.com/scverse/napari-spatialdata/issues/66. This is based of the code from this PR: https://github.com/scverse/napari-spatialdata/pull/93.

### CODE REVIEW

**Comment by LucaMarconato** [comment]
@rahulbshrestha is the PR ready for review (except for the test to be fixed)?

**Comment by rahulbshrestha** [comment]
@LucaMarconato I'll add one test for transform_to_rgb function in test_utils. Then, I'll ask for your review.

**Comment by melonora** [comment]
@LucaMarconato @rahulbshrestha should we extend blobs with 'is_3d' and 'is_stack'?

**Comment by melonora** [comment]
This PR [#307](https://github.com/scverse/spatialdata/pull/307) is a prerequisite for the test added here.

**Comment by melonora** [comment]
@LucaMarconato @kevinyamauchi This one can be reviewed once the PR in SpatialData is merged.

**Comment by kevinyamauchi** [comment]
@melonora , I just cut a new release with https://github.com/scverse/spatialdata/pull/307. It should be on pypi in a few minutes. Can you please re-run the tests and then ping me if this PR is ready for review? Thanks!

File: src/napari_spatialdata/utils/_utils.py [hunk 1]
```diff
@@ -221,3 +225,56 @@ def _points_inside_triangles(points: NDArrayA, triangles: NDArrayA) -> NDArrayA:
...
+    into c x y x x and flags as suitable for rgb visualization.
+
+    Parameters
+    ----------
+    element: SpatialImage | MultiScaleSpatialImage
+        Element in sdata.images
+    Returns
+    -------
+    new_raster: DataArray
+        The image in shape of c x y x x.
```
Comment (kevinyamauchi) [inline src/napari_spatialdata/utils/_utils.py:243]:
```suggestion
        The image in shape of (c, y, x)
```

File: src/napari_spatialdata/utils/_utils.py [hunk 2]
```diff
@@ -221,3 +225,56 @@ def _points_inside_triangles(points: NDArrayA, triangles: NDArrayA) -> NDArrayA:
         out[i] = _point_inside_triangles(triangles - points[i])
 
     return out
+
+
+def _transform_to_rgb(element: SpatialImage | MultiscaleSpatialImage) -> tuple[DataArray, bool]:
+    """Swap the axes to y, x, c if an image supports rgb(a) visualization.
+
+    Checks whether c dim is present in the axes and allows for rgb(a) visualization. If so, subsequently transposes it
+    into c x y x x and flags as suitable for rgb visualization.
```
Comment (kevinyamauchi) [inline src/napari_spatialdata/utils/_utils.py:234]:
```suggestion
    into (c, y, x) axis order and flags as suitable for rgb visualization.
```

File: src/napari_spatialdata/_interactive.py [hunk 3]
```diff
@@ -213,8 +213,10 @@ def _add_label(self, key: str) -> None:
         selected_cs = self.coordinate_system_widget._system
         affine = _get_transform(self._sdata.labels[key], selected_cs)
 
+        rgb_labels, rgb = _transform_to_rgb(element=self._sdata.labels[key])
```
Comment (kevinyamauchi) [inline src/napari_spatialdata/_interactive.py:216]: I find this function name a bit confusing in this context as we don't want to transform the labels to RGB. Not a major blocker, but is there a more clear name we can use?

Also, I assume we aren't using the `rgb` flag here. If that is the case, I suggest (minor nitpick):
```suggestion
        rgb_labels, _= _transform_to_rgb(element=self._sdata.labels[key])
```

File: src/napari_spatialdata/_interactive.py [hunk 4]
```diff
@@ -232,11 +234,14 @@ def _add_image(self, key: str) -> None:
         selected_cs = self.coordinate_system_widget._system
         img = self._sdata.images[key]
         affine = _get_transform(self._sdata.images[key], selected_cs)
+        new_image, rgb = _transform_to_rgb(element=self._sdata.images[key])
```
Comment (kevinyamauchi) [inline src/napari_spatialdata/_interactive.py:237]: minor nitpick: I don't find this variable name to be very clear. Perhaps use the same pattern as above in the add label method (i.e., `rgb_image`)

**Review by kevinyamauchi — APPROVED** [review]
Thanks @rahulbshrestha and @melonora ! I think this looks good. I left a couple of minor comments below. I think we can merge once you address those.

File: src/napari_spatialdata/_interactive.py (hunk 4 above)
Comment (melonora) [reply src/napari_spatialdata/_interactive.py]: Adressed with [b43f9b2](https://github.com/scverse/napari-spatialdata/pull/115/commits/b43f9b232f6486d5dba740dc2342a86c9de768a0)

**Comment by melonora** [comment]
> Thanks @rahulbshrestha and @melonora ! I think this looks good. I left a couple of minor comments below. I think we can merge once you address those.

@kevinyamauchi I think everything has been addressed:)

**Pull request merged**